testing/tests/DevExpress.ui.widgets.form/form.tests.js

```diff
@@ -1727,6 +1730,126 @@ QUnit.test('Labels are not aligned when labelLocation is top', function(assert)
+            items: [
+                {
+                    itemType: 'group',
+                    items: [
+                        {
+                            itemType: 'group',
+                            items: [{
+                                dataField: editorType,
+                                label: { visible: false },
+                                editorType: editorType,
+                            }, ],
+                        },
+                    ],
+                },
+
+            ],
+        }).dxForm('instance');
+
+        const $componentWrapper = testContainer.find(`.dx-${selector}`).parent();
+
+        assert.notOk($componentWrapper.hasClass(TOGGLE_CONTROLS_PADDING_CLASS));
+    });
+
+    ['left', 'right'].forEach(labelLocation => {
+        QUnit.test(`${editorType} should not have a class when the labelLocation = ${labelLocation} (T1126956)`, function(assert) {
+            const testContainer = $('#form');
+
+            testContainer.dxForm({
+                labelLocation,
+                items: [
+                    {
+                        itemType: 'group',
+                        items: [
+                            {
+                                itemType: 'group',
+                                items: [{
+                                    dataField: editorType,
+                                    label: { visible: true },
+                                    editorType: editorType,
+                                }, ],
+                            },
+                        ],
+                    },
+
+                ],
+            }).dxForm('instance');
+
+            const $componentWrapper = testContainer.find(`.dx-${selector}`).parent();
+
+            assert.notOk($componentWrapper.hasClass(TOGGLE_CONTROLS_PADDING_CLASS));
+        });
+    });
+
+    ['center', 'right'].forEach(alignment => {
+        QUnit.test(`${editorType} should not have a class when the alignment = ${alignment} (T1126956)`, function(assert) {
+            const testContainer = $('#form');
+
+            testContainer.dxForm({
+                labelLocation: 'top',
+                items: [
+                    {
+                        itemType: 'group',
+                        items: [
+                            {
+                                itemType: 'group',
+                                items: [{
+                                    dataField: editorType,
+                                    label: { visible: true, alignment },
+                                    editorType: editorType,
+                                }, ],
+                            },
+                        ],
+                    },
+
+                ],
+            }).dxForm('instance');
+
+            const $componentWrapper = testContainer.find(`.dx-${selector}`).parent();
+
+            assert.notOk($componentWrapper.hasClass(TOGGLE_CONTROLS_PADDING_CLASS));
+        });
+    });
+});
+
```

Comment:
Does label toggle on a `label.visible` / `label.alignment` / `label.location` / `editorType` option change?

Reply:
if you meant class toggle, then yes, it works with different switching of these options, and only when all these options are equal to certain values do we set the class